### PR TITLE
feat(FrameManage): add title for WaitTask

### LINKED ISSUE #2037: Improve error message for timeouts
When a `page.waitFor()` times out, this is all error context a dev gets:
```
     Error: waiting failed: timeout 30000ms exceeded
      at Timeout.WaitTask._timeoutTimer.setTimeout (node_modules/puppeteer/lib/FrameManager.js:695:58)
```

No stack trace and no indication _what_ operation timed out.

An easy change to an error message like
```
     Error: waiting for selector ".my-selector" failed: timeout 30000ms exceeded
      at Timeout.WaitTask._timeoutTimer.setTimeout (node_modules/puppeteer/lib/FrameManager.js:695:58)
```

would make a huge difference in knowing where to look.

### PR DESCRIPTION
**Pull request by yanivefraim**
Closes #2037 
(no answer from @felixfbecker regarding https://github.com/GoogleChrome/puppeteer/pull/2084. Created a new PR. Sorry for that...)

### CODE REVIEW

File: lib/FrameManager.js [hunk 1]
```diff
@@ -696,7 +696,8 @@ class Frame {
     const waitForVisible = !!options.visible;
     const waitForHidden = !!options.hidden;
     const polling = waitForVisible || waitForHidden ? 'raf' : 'mutation';
-    return this.waitForFunction(predicate, {timeout: options.timeout, polling}, selectorOrXPath, isXPath, waitForVisible, waitForHidden);
+    const timeout = helper.isNumber(options.timeout) ? options.timeout : 30000;
```
Comment (aslushnikov) [inline lib/FrameManager.js:699]: is this a drive-by? Looks like an unrelated bugfix.

Comment (yanivefraim) [reply lib/FrameManager.js:699]: Actually it is related to this comment:

https://github.com/GoogleChrome/puppeteer/pull/2084#comment-170343800

Comment (aslushnikov) [reply lib/FrameManager.js:699]: Your comment link didn't work for me.

But if I understand correctly, the `page.waitForSelector`/`page.waitForXPath` don't currently have default timeout 30 seconds (despite documentation claiming so), and this change brings it back.

**Review by aslushnikov — APPROVED** [review]
Nice patch

File: lib/FrameManager.js (hunk 1 above)
Comment (yanivefraim) [reply lib/FrameManager.js:699]: sorry, created it from mobile (it is working on mobile... strange)

The comment was: https://github.com/GoogleChrome/puppeteer/pull/2084#discussion_r170343800

Comment (aslushnikov) [reply lib/FrameManager.js:699]: ah I see now. Thank you for the explanation.

**Comment by felixfbecker** [comment]
Thank you!
